src/App/BlogPost/Presentation/Controller/EditAction.php

```diff
@@ -0,0 +1,28 @@
+<?php
+
+namespace Solid\App\BlogPost\Presentation\Controller;
+
+use Solid\App\BlogPost\Application\UseCase\EditBlogPostUseCase;
+use Solid\App\BlogPost\Presentation\Renderer\Renderer;
+
+final class EditAction
+{
+    public function __construct(
+        private readonly EditBlogPostUseCase $useCase,
+        private readonly Renderer $renderer,
+    ) {
+    }
+
+    public function __invoke(int $id, ?string $author, ?string $title, ?string $content, string $format = 'html'): void
+    {
+        $blogPost = $this->useCase->execute($id, $author, $title, $content);
+
+        if ('json' === $format) {
```

Comment:
[OCP] This is open for modification. An extension mechanism would be needed to avoid modifying this class each time a new format is added.